src/mono/mono/metadata/bundled-resources-internals.h

```diff
@@ -0,0 +1,105 @@
+// Licensed to the .NET Foundation under one or more agreements.
```

Comment:
Instead of mixing glib types and C99 types in this API that potentially could end up as a public API  in the end, I thikn we should only use C99 types, we should also drop dependencies on MonoBundledAssembly and MonoBundledSatelliteAssembly in this header. Since we cut combability on old struct layout we should redefine our own versions here as part of the bundling API, they should probably still be 1:1 with the old, so:

typedef struct _MonoBundledAssemblyData {
	const char *name;
	const uint8_t *data;
        uint32_t size;
} MonoBundledAssemblyData;

typedef struct _MonoBundledSatelliteAssemblyData {
	const char *name;
        const char *culture;
	const uint8_t *data;
        uint32_t size;
} MonoBundledAssemblyData;

Reply:
Good idea, I dropped all internal types and introduced the parallels of `MonoBundledAssembly` and `MonoBundledSatelliteAssembly` as `MonoBundledAssemblyData` and `MonoBundledSatelliteAssemblyData`.